internal/backends/sqlite/collection.go

```diff
@@ -211,8 +211,50 @@ func (c *collection) DeleteAll(ctx context.Context, params *backends.DeleteAllPa
 
 // Explain implements backends.Collection interface.
 func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams) (*backends.ExplainResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3050
-	panic("not implemented")
+	db := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if db == nil {
+		return &backends.ExplainResult{
+			QueryPlanner: must.NotFail(types.NewDocument()),
+		}, nil
+	}
+
+	meta := c.r.CollectionGet(ctx, c.dbName, c.name)
+	if meta == nil {
+		return &backends.ExplainResult{
+			QueryPlanner: must.NotFail(types.NewDocument()),
+		}, nil
+	}
+
+	q := fmt.Sprintf(`EXPLAIN QUERY PLAN SELECT %s FROM %q`, metadata.DefaultColumn, meta.TableName)
+
+	rows, err := db.QueryContext(ctx, q)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	defer rows.Close()
+
+	queryPlan, err := types.NewArray()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	for rows.Next() {
+		var id int32
+		var parent int32
+		var notused int32
+		var detail string
+
+		if err := rows.Scan(&id, &parent, &notused, &detail); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		queryPlan.Append(fmt.Sprintf("id=%d parent=%d notused=%d detail=%s", id, parent, notused, detail))
```

Comment:
Does the output appear sensible?

Reply:
As far as I understand, explain for SQLite is a bit different than what we see for PostgreSQL. So, in addition to the plan itself, they also have those extra fields that show the relations between steps in the plan. I just added the simplest approach (we print all the fields that are returned). But maybe there is a better idea of what we can do with it. We don't have complicated plans now (as we query everything, no WHERE clauses, no indexes etc), but maybe in the future we have a better idea how to print it.

Reply:
That's fine. I'm not familiar with SQLite execution plan so I just wanted to ask above is what we want to return :hugs:

For now, having something is good. But I'm lacking what they mean, perhaps just adding comment to link where we can find more info about id, parent etc would be good :hugs:

Reply:
Yeah, let's add it. I used this docs - https://www.sqlite.org/eqp.html - to understand it.

> A query plan is represented as a tree. In raw form, as returned by sqlite3_step(), each node of the tree consists of four fields: An integer node id, an integer parent id, an auxiliary integer field that is not currently used, and a description of the node.

The tricky part is that SQLite cli tools can draw quite nice plans based on this info (basically, like a tree where parents-children look quite nice), but implementing it on our side looks unnecessary for now. And also some other clients (like the one in Goland) don't show it as a tree, but just print these exact four columns... So I think we are ok for now.

Reply:
Added!

Reply:
We don't have to worry about prettiness for now :hugs:
I think `id` and `detail` gives idea of what's going on which is nice!

```
test> db.collection.insertMany([{_id: 1}, {_id: 2}])

test> db.runCommand({explain: { find: "colls", query: { _id: 1 } }})
{
  queryPlanner: { Plan: [ 'id=2 parent=0 notused=0 detail=SCAN colls_367c18bc' ] },
  explainVersion: '1',
  command: { find: 'colls', query: { _id: 1 }, '$db': 'test' },
  serverInfo: {
    host: 'alma',
    version: '6.0.42',
    gitVersion: 'c6a73ee1f1f5c235b4464c2e9dadf9f033747e17',
    ferretdbVersion: 'v1.9.0-30-gc6a73ee1'
  },
  pushdown: false,
  sortingPushdown: false,
  limitPushdown: false,
  ok: 1
}
```